### ISSUE TITLE
2nd chapter code AMI does no longer exist

### ISSUE THREAD

**Issue by aanzolaavila** [brikis98/terraform-up-and-running-code#68]
On https://github.com/brikis98/terraform-up-and-running-code/blob/master/code/terraform/02-intro-to-terraform-syntax/webserver-cluster/main.tf, the AMI for the instances does no longer exist and fails, and I don't know what other image would contain busybox for trying out the HTTP requests

I was trying out **ami-0c55b159cbfafe1f0**, but it does not have busybox installed.

**Comment by brikis98** [comment]
I'm pretty sure `ami-0c55b159cbfafe1f0` still exists, but AMIs are region-specific, so that's the ID solely for the `us-east-2` region. Make sure that's the region you pick!

**Comment by aanzolaavila** [comment]
I was using **us-east-1**, but I think I found a better solution for me on https://learn.hashicorp.com/tutorials/terraform/cloud-init, in which the `main.tf` file in the **instances** folder contains this config, which will get the latest ubuntu image without having to specify a specific AMI, which for me is a better solution that is independent from the region as you pointed out.

```
resource "aws_launch_configuration" "example" {
  image_id = data.aws_ami.ubuntu.id
  instance_type = "t2.micro"
  security_groups = [aws_security_group.instance.id, aws_security_group.ssh.id]

  user_data = data.template_file.user_data.rendered

  lifecycle { 
    create_before_destroy = true
  }
}

data "aws_ami" "ubuntu" {
  most_recent = true

  filter {
    name = "name"
    values = ["ubuntu/images/hvm-ssd/ubuntu-*20*-amd64-server-*"]
  }

  filter {
    name = "virtualization-type"
    values = ["hvm"]
  }
  
  owners = ["099720109477"] # Canonical
}
```

**Comment by brikis98** [comment]
Yup, keep reading, we use the `aws_ami` data source later in the book :)

**Comment by Mr-Slippery** [comment]
In case you are also stuck on trying the web server on a different region, this is how to find out the right AMI Id for your region. Search the given AMI Id on region `us-east-2` and find the AMI Name (which might be worth mentioning in a possible future edition of the book too):
![image](https://user-images.githubusercontent.com/5954346/119261689-ce18e480-bbd8-11eb-9f59-3b0fcf8723b4.png)
And then switch to the region you deploy to and use the same search interface but search for the AMI Name: `ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-20190212.1`
Hope this helps.

**Comment by brikis98** [comment]
See also: https://blog.gruntwork.io/locating-aws-ami-owner-id-and-image-name-for-packer-builds-7616fe46b49a

**Comment by mikerossiter** [comment]
Hi guys. I'm in a similar situation. I have tried these suggestions but the web server doesn't seem to want to start. Any ideas?

**Comment by Gang-Peng** [comment]
yup that ami is not there anymore for us-east-2, the closest I can find is this one: ami-0e3606a8091a015e5
![image](https://github.com/brikis98/terraform-up-and-running-code/assets/7462807/255805f8-231a-46f3-85a0-e6976c0032e2)

**Comment by Gang-Peng** [comment]
I cannot find that image (ami-0c55b159cbfafe1f0) anymore in us-east-2. The closest one is "ami-03a5def6b0190cef7".
![image](https://github.com/brikis98/terraform-up-and-running-code/assets/7462807/90cc5790-c114-4fe4-bccb-996b9eff80e3)

**Comment by temutai** [comment]
> I cannot find that image (ami-0c55b159cbfafe1f0) anymore in us-east-2. The closest one is "ami-03a5def6b0190cef7". ![image](https://private-user-images.githubusercontent.com/7462807/296521999-90cc5790-c114-4fe4-bccb-996b9eff80e3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE2MDM3MzgsIm5iZiI6MTcyMTYwMzQzOCwicGF0aCI6Ii83NDYyODA3LzI5NjUyMTk5OS05MGNjNTc5MC1jMTE0LTRmZTQtYmNjYi05OTZiOWVmZjgwZTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjFUMjMxMDM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzQzZTY4YTdmZjgwMGIzZDRkYzJkMTdlZDE3MWIxZGZkNDA4M2RiNGI3ZTRmMjViYjc4ZmZlN2MzMDFkMzIzMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.cX1-B6Vdce14fCpfeNTC9c84jkvNE8_fEkymLjTxVOs)

Pick any available free tier Ubuntu instance. Also it doesn't have to be us-east-2. I did it in us-east-1
